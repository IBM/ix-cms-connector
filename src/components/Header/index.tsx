import { FunctionComponent } from "preact";

/* Example of function component */
const MainHeader: FunctionComponent<{title: string}> = ({title /*, children*/}) => {
  return (
    <header class="text-center py-2 bg-slate-200 mb-4">
      <h1 class="text-2xl">LIT</h1>
      <h2>{title}</h2>
    </header>
  );
} ;

export default MainHeader;