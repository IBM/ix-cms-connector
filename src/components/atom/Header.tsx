import { FunctionComponent } from "preact";

/* Example of function component */
export const Header: FunctionComponent = ({ children }) => {
  return (
    <header class="flex justify-start items-center bg-ui-05 px-2">
      <div class="text-xl text-text-04 p-2">
        IBM <strong>iX</strong>
      </div>
      {!!children && <div>{children}</div>}
    </header>
  );
};
