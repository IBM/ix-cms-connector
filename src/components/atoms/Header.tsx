/*
 * Copyright 2020- IBM Inc. All rights reserved
 * SPDX-License-Identifier: Apache2.0
 */
import { FunctionComponent } from "preact";

export const Header: FunctionComponent = ({ children }) => {
  return (
    <header
      class="flex justify-start items-center bg-ui-05 p-4"
      data-testId="header-component"
    >
      <div class="text-xl text-text-04">
        IBM <strong>iX</strong>
      </div>
      {!!children && <div>{children}</div>}
    </header>
  );
};
