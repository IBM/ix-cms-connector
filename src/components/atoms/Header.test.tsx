/*
 * Copyright 2020- IBM Inc. All rights reserved
 * SPDX-License-Identifier: Apache2.0
 */
// @vitest-environment jsdom

import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/preact";

import { Header } from "./Header";

describe("Header", () => {
  it("should be rendered on the screen", () => {
    render(<Header />);

    const headerEl = screen.getByTestId("header-component");
    expect(headerEl).toBeInTheDocument();
  });

  it("should match the existing snapshot", () => {
    render(<Header />);

    const headerEl = screen.getByTestId("header-component");
    expect(headerEl).toMatchSnapshot();
  });
});
