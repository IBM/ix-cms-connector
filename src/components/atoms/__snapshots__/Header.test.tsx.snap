// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match the existing snapshot 1`] = `
<header
  class="flex justify-start items-center bg-ui-05 p-4"
  data-testid="header-component"
>
  <div
    class="text-xl text-text-04"
  >
    IBM 
    <strong>
      iX
    </strong>
  </div>
</header>
`;
